shell/platform/fuchsia/flutter/component.cc

```diff
@@ -69,6 +82,30 @@ static std::string DebugLabelForURL(const std::string& url) {
   }
 }
 
+static std::unique_ptr<fml::FileMapping> make_read_exec_mapping(
+    const char* path) {
+  int fd;
+  zx_status_t status =
+      fdio_open_fd(path, OPEN_RIGHT_READABLE | OPEN_RIGHT_EXECUTABLE, &fd);
+
+  if (status != ZX_OK) {
+    return nullptr;
+  }
+
+  using Protection = fml::FileMapping::Protection;
+
+  auto mapping = std::make_unique<fml::FileMapping>(
+      fml::UniqueFD{FML_HANDLE_EINTR(fd)},
```

Comment:
`FML_HANDLE_EINTR` retries the statement in case of `EINTR`. Two issues here:

1: On Fuchsia, there is no EINTR.
2: If Fuchsia had EINTR, the statement that was repeated would no attempt to change the assignment leading to an infinite loop :( (there is a limit in `NDEBUG` but still)

Let's just get rid of FML_HANDLE_EINTR here but the second bit of feedback is a heads up for general use of the EINTR checker macro on non-Fuchsia platforms.

Reply:
Can you add a comment here saying that `::close` is fine on handles opened with `::fdio_open_fd`? It is not immediately clear that this is verified behavior. Otherwise, we will be leaking handles.

Reply:
My comment about the EINTR should have mentioned that the macro should have wrapped the statement that created the FD in the first place, so something like `FML_HANDLE_EINTR(..open..)`.

Reply:
removing the EINTR macro for now. Not retrying on Fuchsia should hopefully be fine. We can revisit it if it turns out to be a problem.